zephyr/src/common.h

```diff
@@ -0,0 +1,10 @@
+
+#define CHECK_SNPRINTF(dest, dest_size, ...) \
+    do { \
+        int result = snprintf(dest, dest_size, __VA_ARGS__); \
```

Comment:
don't declare variables in macros, let functions declare them and we just use it here